### ISSUE TITLE
File with bad EXIF stop uploader without error message

### ISSUE THREAD

**Issue by luxigo** [doxel/doxel-angular#49]

**Comment by luxigo** [comment]
https://github.com/doxel/doxel-angular/commit/3c78626c4ca2f1df37b9ac6909d635faa20e12d2 is broken...

Looks like $q is not appropriate here because in the example below resolvedValue is still undefined after deffered.resolve(123)

We must switch to Q.

```
it('should simulate promise', inject(function($q, $rootScope) {
  var deferred = $q.defer();
  var promise = deferred.promise;
  var resolvedValue;

  promise.then(function(value) { resolvedValue = value; });
  expect(resolvedValue).toBeUndefined();

  // Simulate resolving of promise
  deferred.resolve(123);
  // Note that the 'then' function does not get called synchronously.
  // This is because we want the promise API to always be async, whether or not
  // it got called synchronously or asynchronously.
  expect(resolvedValue).toBeUndefined();

  // Propagate promise resolution to 'then' functions using $apply().
  $rootScope.$apply();
  expect(resolvedValue).toEqual(123);
}));
```

**Comment by luxigo** [comment]
fixed in commit 9e182ed975c0ca2ad0634a5ce2e091a7635d2efa
